central/externalbackups/plugins/s3compatible/s3compatible.go

```diff
@@ -0,0 +1,238 @@
+package s3compatible
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	s3Types "github.com/aws/aws-sdk-go-v2/service/s3/types"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/externalbackups/plugins"
+	"github.com/stackrox/rox/central/externalbackups/plugins/types"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/errorhelpers"
+	"github.com/stackrox/rox/pkg/httputil/proxy"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/urlfmt"
+)
+
+const (
+	backupMaxTimeout               = 4 * time.Hour
+	testMaxTimeout                 = 5 * time.Second
+	initialConfigurationMaxTimeout = 5 * time.Minute
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+func init() {
+	plugins.Add(types.S3CompatibleType, func(backup *storage.ExternalBackup) (types.ExternalBackup, error) {
+		return newS3Compatible(backup)
+	})
+}
+
+type s3Compatible struct {
+	integration *storage.ExternalBackup
+
+	client   *s3.Client
+	uploader *manager.Uploader
+}
+
+func validate(cfg *storage.S3Compatible) error {
+	errorList := errorhelpers.NewErrorList("S3 Compatible Validation")
+	if !cfg.GetUseIam() {
```

Comment:
I would suggest to remove the container IAM role from S3 compatible providers. Reason being that while some providers may provide a similar feature (see minio for example https://min.io/docs/minio/linux/developers/security-token-service/AssumeRoleWithWebIdentity.html#minio-sts-assumerolewithwebidentity), it will require more config like a custom STS endpoint. So I think it's better to leave this feature as AWS only for now to keep documentation and configuration simpler. We can add this back if somebody asks us for it.

Reply:
Makes sense. I've removed it 👍